### PR TITLE
Added worker arguments to python_test to allow python_test to accept worker scripts

### PR DESCRIPTION
**Pull request by BNMetrics**

### CODE REVIEW

File: src/parse/rules/python_rules.build_defs [hunk 1]
```diff
@@ -248,6 +249,11 @@ def python_test(name:str, srcs:list, data:list=[], resources:list=[], deps:list=
         )
         deps = [pex_rule, lib_rule]
 
+    test_cmd = '$TEST ' + flags
+    if worker:
+        test_cmd += '$(worker %s) && ' % worker
```
Comment (agenticarus) [inline src/parse/rules/python_rules.build_defs]: Won't this put it on the end of the string instead of the start?

**Comment by peterebden** [comment]
Nice, but I think we should do the same to the other `_test` rules as well for consistency. Do you mind having a look at those too please?

File: src/parse/rules/cc_rules.build_defs [hunk 2]
```diff
@@ -477,11 +478,12 @@ def cc_test(name:str, srcs:list=[], hdrs:list=[], compiler_flags:list&cflags&cop
         )
         srcs.append(main_rule)
     cmds, tools = _binary_cmds(_c, linker_flags, pkg_config_libs)
-    test_cmd = None if not CONFIG.CPP_COVERAGE else {
-        'opt': '$TEST %s' % flags,
-        'dbg': '$TEST %s' % flags,
-        'cover': '$TEST %s; R=$?; cp $GCNO_DIR/*.gcno . && gcov *.gcda && cat *.gcov > test.coverage; exit $R' % flags,
-    }
+    # test_cmd = None if not CONFIG.CPP_COVERAGE else {
```
Comment (peterebden) [inline src/parse/rules/cc_rules.build_defs]: delete these lines

File: src/parse/rules/cc_rules.build_defs [hunk 3]
```diff
@@ -495,6 +497,17 @@ def cc_test(name:str, srcs:list=[], hdrs:list=[], compiler_flags:list&cflags&cop
             _c=_c,
         )
         deps.append(lib_rule)
+
+    test_cmd = '$TEST' % flags
```
Comment (peterebden) [inline src/parse/rules/cc_rules.build_defs]: this is missing any symbol to interpolate?

File: src/parse/rules/cc_rules.build_defs [hunk 4]
```diff
@@ -495,6 +497,17 @@ def cc_test(name:str, srcs:list=[], hdrs:list=[], compiler_flags:list&cflags&cop
...
+    if worker:
+        test_cmd = '$(worker %s) && %s ' % (worker, test_cmd)
+        deps += [worker]
+
+    test_cmds = {k: test_cmd for k in ['opt', 'dbg']}
```
Comment (peterebden) [inline src/parse/rules/cc_rules.build_defs]: nit: move this inside the if statement like
```
if CONFIG.CPP_COVERAGE:
    test_cmd = {
        'opt': test_cmd,
        'dbg': test_cmd,
        'cover': test_cmd + '...',
    }
```

File: src/parse/rules/sh_rules.build_defs [hunk 5]
```diff
@@ -110,6 +111,7 @@ def sh_test(name:str, src:str=None, labels:list&features&tags=None, data:list=No
         srcs=[src or test],
         data=data,
         deps=deps,
+        worker=worker,
```
Comment (peterebden) [inline src/parse/rules/sh_rules.build_defs]: I don't think you can do this? `build_rule` doesn't take `worker` directly?

File: src/parse/rules/cc_rules.build_defs (hunk 3 above)
Comment (BNMetrics) [reply src/parse/rules/cc_rules.build_defs]: oops! Will fix this now!

File: src/parse/rules/sh_rules.build_defs (hunk 5 above)
Comment (BNMetrics) [reply src/parse/rules/sh_rules.build_defs]: More oops >.<

**Pull request merged**